internal/tutils/psql.go

```diff
@@ -0,0 +1,136 @@
+package tutils
+
+import (
+	"fmt"
+	"math/rand"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/osbuild/image-builder/internal/db"
+)
+
+type PSQLContainer struct {
+	name string
+	id   string
+	port int
+}
+
+const (
+	image string = "postgres:13"
+	user  string = "postgres"
+)
+
+func containerRuntime() (string, error) {
+	out, err := exec.Command("which", "podman").Output()
+	if err == nil {
+		return strings.TrimSpace(string(out)), nil
+	}
+	out, err = exec.Command("which", "docker").Output()
+	if err == nil {
+		return strings.TrimSpace(string(out)), nil
+	}
+	return "", err
+}
+
+func NewPSQLContainer() (*PSQLContainer, error) {
+	rt, err := containerRuntime()
+	if err != nil {
+		return nil, err
+	}
+
+	name := fmt.Sprintf("image_builder_test_%d", time.Now().Unix())
+	rand.Seed(time.Now().Unix())
+	/* #nosec G404 */
+	port := 65535 - rand.Intn(32000)
+	/* #nosec G204 */
+	out, err := exec.Command(
+		rt,
+		"run",
+		"--detach",
+		"--rm",
+		"--name", name,
+		"--env", fmt.Sprintf("POSTGRES_USER=%s", user),
+		"--env", "POSTGRES_HOST_AUTH_METHOD=trust",
+		"-p", fmt.Sprintf("127.0.0.1:%d:5432", port),
+		image,
+	).Output()
+	if err != nil {
+		fmt.Println(out, err)
+		return nil, err
+	}
+
+	p := &PSQLContainer{
+		name: name,
+		id:   strings.TrimSpace(string(out)),
+		port: port,
+	}
+
+	tries := 0
+	for tries < 10 {
+		_, err := p.execCommand([]string{
+			"exec",
+			p.name,
+			"pg_isready",
+		})
+		if err != nil {
+			time.Sleep(time.Second * 1)
+			continue
+		}
+		break
+	}
+	return p, nil
+}
+
+func (p *PSQLContainer) execCommand(args []string) (string, error) {
```

Comment:
nitpick: `args ...string` would make the API similar to the one `exec.Command` has.